statsmodels/tsa/statespace/tvp.py

```diff
@@ -0,0 +1,140 @@
+"""
+Time varying parameters model
+
+Author: Valery Likhosherstov
+License: Simplified-BSD
+"""
+import numpy as np
+from .mlemodel import MLEModel
+
+
+class TVPModel(MLEModel):
+    r"""
+    Time varying parameters model
+
+    Parameters
+    ----------
+    endog : array_like
+        The observed time series process :math:`y`.
+    exog : array_like, optional
+        Array of exogenous regressors for the observation equation, shaped
+        nobs x k_exog. Required for this model.
+    kwargs
+        Keyword arguments, passed to superclass (`MLEModel`) initializer.
+
+    Attributes
+    ----------
+    k_tv_params : int
+        The number of time varying parameters.
+
+    Notes
+    -----
+    The specification of the model is as follows:
+
+    .. math::
+
+        y_t = \beta_{1t} x_{1t} + \beta_{2t} x_{2t} + ... +
+        \beta_{kt} x_{kt} + e_t \\
+        \beta_{it} = \beta_{i,t-1} + v_{it} \\
+        e_t \sim N(0, \sigma^2) \\
+        v_{it} \sim N(0, \sigma_i^2)
+
+    where :math:`\beta_{it}` are time varying parameters. The number of
+    time varying parameters (:math:`k`) is defined by `exog` second dimension.
+
+    References
+    ----------
+    Kim, Chang-Jin, and Charles R. Nelson. 1999.
+    "State-Space Models with Regime Switching:
+    Classical and Gibbs-Sampling Approaches with Applications".
+    MIT Press Books. The MIT Press.
+    """
+
+    def __init__(self, endog, exog=None, **kwargs):
+
+        # Check if exogenous data is provided
+        if exog is None:
+            raise ValueError('Exogenous data is required for this model.')
+
+        # Transform to numpy array
+        exog = np.asarray(exog)
+
+        # Reshape exog, if it is one-dimensional array
+        if exog.ndim == 1:
+            exog = exog.reshape(1, -1)
+
+        # The number of time varying parameters is defined by `exog` second
+        # dimension
+        self.k_tv_params = exog.shape[1]
+        k_tv_params = self.k_tv_params
+
+        # Superclass initialization
+        super(TVPModel, self).__init__(endog, k_tv_params, exog=exog, **kwargs)
+
+        # Every observation is a single value
+        if self.k_endog != 1:
+            raise ValueError('Endogenous vector must consist of 1 value.')
```

Comment:
This might be a misleading error message (when I read it, I thought you meant that `endog` had to be a constant value); maybe use: 'Endogenous vector must be univariate'.